homeassistant/components/geofency/__init__.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for Geofency.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/geofency/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import see
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import HTTP_UNPROCESSABLE_ENTITY, STATE_NOT_HOME, \
+    ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'geofency'
+DEPENDENCIES = ['http']
+
+CONF_MOBILE_BEACONS = 'mobile_beacons'
+
+CONFIG_SCHEMA = vol.Schema({
+    vol.Optional(DOMAIN): vol.Schema({
+        vol.Optional(CONF_MOBILE_BEACONS, default=[]): vol.All(
+            cv.ensure_list,
+            [cv.string]
+        ),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+ATTR_CURRENT_LATITUDE = 'currentLatitude'
+ATTR_CURRENT_LONGITUDE = 'currentLongitude'
+
+BEACON_DEV_PREFIX = 'beacon'
+CONF_MOBILE_BEACONS = 'mobile_beacons'
+
+LOCATION_ENTRY = '1'
+LOCATION_EXIT = '0'
+
+URL = '/api/geofency'
+
+TRACKER_UPDATE = 'tracker_update'
+
+
+async def async_setup(hass, config):
+    """Set up the Geofency component."""
+    config = config[DOMAIN]
+    mobile_beacons = config[CONF_MOBILE_BEACONS]
+    hass.data[DOMAIN] = [slugify(beacon) for beacon in mobile_beacons]
+    hass.http.register_view(GeofencyView(see, hass.data[DOMAIN]))
+
+    await async_load_platform(hass, 'device_tracker', DOMAIN)
+    return True
+
+
+class GeofencyView(HomeAssistantView):
```

Comment:
Any reason we wouldn't combine this right away with the webhook conversion?

Reply:
I wanted to make it easier to read and review by not making too many sweeping changes at once.

My general approach is one isolated change at a time otherwise things tend to get messy. I consider this prerequisite work (regardless of whether we did the webhook conversion or not) so I think it can stand alone.

Reply:
Makes sense. I just would want to avoid not getting the 2nd PR in before the release and then having 2 subsequent breaking changes (we'll get pitchforks 😉)

Reply:
Haha, yes agreed!